tests/models/test_baseoperator.py

```diff
@@ -248,6 +249,13 @@ def test_custom_resources(self):
         self.assertEqual(task.resources.cpus.qty, 1)
         self.assertEqual(task.resources.ram.qty, 1024)
 
+    @conf_vars({('email', 'default_email_on_retry'): 'True',
+                ('email', 'default_email_on_failure'): 'False'})
+    def test_default_email_on_actions(self):
+        test_task = DummyOperator(task_id='test_default_email_on_actions')
+        assert test_task.email_on_retry is True
+        assert test_task.email_on_failure is False
```

Comment:
Hi @kaxil, needs some help. When I set `('email', 'default_email_on_failure'): 'False'`, it seems that `test_task.email_on_failure` is still showing as True. 

While when I tried to directly get the config `conf.getboolean('email', 'default_email_on_failure')`, it is actually set as `False`. I cannot tell what's going on.... Do you have any clues?

Reply:
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments 
Default arguments are cached, so you have to reload the module
Example:
```
    @conf_vars({("core", "executor"): "CeleryExecutor"})
    def setUpClass(cls):
        importlib.reload(cli)
        cls.parser = cli.CLIFactory.get_parser()
```

Reply:
Oh I see! @mik-laj Thanks for the help. I will fix it and update the PR